statarb fixes

bogdanKaftanatiy opened the pull request:
Separated sync and async cancel order operations.
Added update order operation. 
Added auto reduce only flag for reduce position orders

Commit:
added sync/async cancel order implementations

Commit:
ccxt: added auto reduce only flag for reducing positions orders

Commit:
added update otder implementation

Commit:
tests fix

bogdanKaftanatiy committed:
Merge remote-tracking branch 'origin/dev' into bohdan/statarb-fixes

bogdanKaftanatiy committed:
ccxt fixes

bogdanKaftanatiy committed:
close position fix

bogdanKaftanatiy committed:
version bump

bogdanKaftanatiy committed:
ccxt instrument parsing fix

bogdanKaftanatiy committed:
tests fix

bogdanKaftanatiy committed:
fix order cancellation

yuriy-arabskyy commented on src/qubx/backtester/broker.py:
```diff
@@ -64,20 +64,66 @@ def send_order_async(
+            )
+
+        # Cancel the existing order first
+        self.cancel_order(order_id)
```
so you decided to always cancel and not use the amend functionality?

yuriy-arabskyy replied:
ah sorry I didn't notice it's simulation

yuriy-arabskyy commented on src/qubx/connectors/ccxt/broker.py:
```diff
@@ -298,8 +312,19 @@ def _prepare_order_payload(
-        if not (reduce_only := options.get("reduceOnly", False)):
+        # Auto-detect if order reduces existing position
+        reduce_only = options.get("reduceOnly", False)
+        if not reduce_only:
```
this is problematic because we could have an opposide side opening order, I think you need to check if the amount by which we enter only decreases exposure and does not open from the other side

yuriy-arabskyy replied:
fixed it

yuriy-arabskyy reviewed (approved):
looks good with some minor comments, only opposide side entry is important to handle properly

yuriy-arabskyy committed:
Merge branch 'main' into bohdan/statarb-fixes

yuriy-arabskyy committed:
Fix auto reduce_only logic to prevent position flips

Ensures reduce_only is only set when order amount doesn't exceed
position size, preventing orders from flipping to opposite side.
